hal/src/nRF52840/i2c_hal.cpp

```diff
@@ -60,8 +60,8 @@ typedef struct {
     uint8_t                     address;    // 7bit data
 
     volatile uint8_t            rx_buf[BUFFER_LENGTH];
-    volatile uint8_t            rx_buf_index;
-    volatile uint8_t            rx_buf_length;
+    uint8_t                     rx_buf_index;
+    uint8_t                     rx_buf_length;
     volatile uint8_t            tx_buf[BUFFER_LENGTH];
```

Comment:
I meant that we don't really need `volatile` for all of `rx_buf`, `rx_buf_index`, `rx_buf_length`, `tx_buf`, `tx_buf_index`, `tx_buf_length`.

Reply:
I see, we use block sending, and the buffers seem not to be optimized. Have removed.